test/unit/admin_mailer_test.rb

```diff
@@ -26,6 +26,28 @@ class AdminMailerTest < ActionMailer::TestCase
     assert email.body.include?(node.latest.render_body_email(request_host))
   end
 
+  test 'notify_comment_moderators' do
+    comment = comments(:first)
+    moderators = User.where(role: %w[moderator admin])
+    assert !moderators.empty?
+
+    assert_difference 'ActionMailer::Base.deliveries.size', 1 do
+      AdminMailer.notify_comment_moderators(comment).deliver_now
+    end
+
+    # # test that it got queued
+    assert !ActionMailer::Base.deliveries.empty?
+    email = ActionMailer::Base.deliveries.last
+
+    assert_not_nil email.to
+    assert_not_nil email.bcc
+    assert_equal ["comment-moderators@#{request_host}"], ActionMailer::Base.deliveries.last.to
+    assert_equal moderators.collect(&:email), ActionMailer::Base.deliveries.last.bcc
+    assert_equal '[New Public Lab poster needs moderation]', email.subject
+    assert email.body.include?("First-time poster <a href='https://#{request_host}/profile/#{comment.author.name}'>#{comment.author.name}</a>
+    has submitted their first research comment!")
+  end
+
```

Comment:
This test is just checking if the email was sent or not when we call mailer. The functional test would check whether the whole feature is in place or not. Like, when a new user generates a comment a mail is sent or not, comment status check. And, mail shouldn't be sent in case of good user and status check.

Regarding GIFs, you can upload them on drive and paste the link here.

(I have also learnt some testing just by working on PublicLab projects :P)

Reply:
Ok, I think I understand. Can you please tell me where I put the functional test in this case? In `comment_mailer` maybe? I have to test the `notify` method, right?

Reply:
You have to test comment creation, as email is sent in this process so test would be including both parts - checking comment creation with appropriate status and Email sent or not. So, `comment_controller_test` seems more good option. Thoughts?